From 6195b52e9ad0d45bd159a96619f91267fd6d71d5 Mon Sep 17 00:00:00 2001
From: Guannan Ren <gren@redhat.com>
Date: Fri, 19 Jul 2013 20:48:09 +0800
Subject: [RHEL-6.5 virt-manager PATCH 1/8] details: the bus, device values of
 USB device is decimal rather than hex

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=820303
(cherry picked from commit 5a316b9a510df5aa0e6a78b6c67ebcc782e8b09a)
Although the code in lookup_nodedev() is general to deal with
hostdev, it has to seperate USB device from PCI device there.
For PCI device, the domain/bus/slot/function is hex.
For USB device, the bus/device is decimal.

This fix makes the label use hostdev pretty_name:

 +------------------------------------------------------+
 | Physical USB Device                                  |
 |   Device: 006:032 RSA RSA SecureID (R) Authenticator |

(crobinso: fix some pylint)
---
 src/virtManager/details.py | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/src/virtManager/details.py b/src/virtManager/details.py
index c88b75e..ee01d60 100644
--- a/src/virtManager/details.py
+++ b/src/virtManager/details.py
@@ -232,19 +232,25 @@ def lookup_nodedev(vmmconn, hostdev):
             return None
         return getattr(node, attr)

-    devtype     = hostdev.type
-    vendor_id   = hostdev.vendor or -1
-    product_id  = hostdev.product or -1
-    device      = intify(hostdev.device, True)
-    bus         = intify(hostdev.bus, True)
-    domain      = intify(hostdev.domain, True)
-    func        = intify(hostdev.function, True)
-    slot        = intify(hostdev.slot, True)
+    devtype = hostdev.type
     found_dev = None

+    vendor_id = product_id = bus = device = \
+        domain = slot = func = None
+
     # For USB we want a device, not a bus
     if devtype == 'usb':
-        devtype = 'usb_device'
+        devtype    = 'usb_device'
+        vendor_id  = hostdev.vendor or -1
+        product_id = hostdev.product or -1
+        bus        = intify(hostdev.bus)
+        device     = intify(hostdev.device)
+
+    elif devtype == 'pci':
+        domain     = intify(hostdev.domain, True)
+        bus        = intify(hostdev.bus, True)
+        slot       = intify(hostdev.slot, True)
+        func       = intify(hostdev.function, True)

     devs = vmmconn.get_nodedevs(devtype, None)
     for dev in devs:
-- 
1.8.3.2

